packages/admin/admin-rte/src/core/Controls/ControlButton.tsx

```diff
@@ -1,11 +1,71 @@
-import { ComponentsOverrides, Theme } from "@mui/material";
+import { ThemedComponentBaseProps } from "@comet/admin";
+import { ComponentsOverrides } from "@mui/material";
+import { css, styled, Theme, useThemeProps } from "@mui/material/styles";
 import { SvgIconProps } from "@mui/material/SvgIcon";
-import { createStyles, WithStyles, withStyles } from "@mui/styles";
 import * as React from "react";
 
 import getRteTheme from "../utils/getRteTheme";
 
-export interface IProps {
+export type RteControlButtonClassKey = "root" | "selected" | "renderAsIcon";
+
+type OwnerState = Pick<IProps, "selected" | "Icon">;
+
+const Root = styled("button", {
+    name: "CometAdminRteControlButton",
+    slot: "root",
+    overridesResolver({ ownerState }: { ownerState: OwnerState }, styles) {
+        return [styles.root, ownerState.selected && styles.selected, ownerState.Icon && styles.renderAsIcon];
```

Comment:
Please only use boolean-checks with the `&&` syntax:

```suggestion
        return [styles.root, ownerState.selected && styles.selected, Boolean(ownerState.Icon) && styles.renderAsIcon];
```

Or just pass a boolean `renderAsIcon` into the `ownerState` instead of `Icon`.